### PR TITLE
fix: make n3 DataFactory extend RDF.DataFactory

### PR DESCRIPTION
**Pull request by jeswr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [NA] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [NA] Provide a URL to documentation or source code which provides context for the suggested changes: N3 implements RDF/JS spec, it is expected that it should extend this interface
- [NA] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/n3/index.d.ts [hunk 1]
```diff
@@ -95,32 +95,36 @@ export class Quad extends BaseQuad implements RDF.Quad {
...
-    function triple<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
+    triple(subject: RDF.Quad_Subject, predicate: RDF.Quad_Predicate, object: RDF.Quad_Object): Quad;
+    triple<Q_In extends RDF.BaseQuad = RDF.Quad, Q_Out extends BaseQuad = Quad>(
         subject: Q_In["subject"],
         predicate: Q_In["predicate"],
         object: Q_In["object"],
     ): Q_Out;
 }
 
+export const DataFactory: DataFactoryInterface;
```
Comment (tpluscode) [inline types/n3/index.d.ts:126]: I wonder if changing a namespace export into an interface + export const won't be an accidental breaking change. I recall looking at this very piece of n3 types but I ran into from issues IIRC

Comment (jeswr) [reply types/n3/index.d.ts:126]: I've run tests across a few of the libs that I commonly use and no issues; will also run on some bigger libs such as Comunica.

@tpluscode would you also be able to test this against any libs you have that you're concerned will break with this change?

Comment (jeswr) [reply types/n3/index.d.ts:126]: Bump @tpluscode - I've also called for people to run tests on codebases they are concerned about in https://github.com/rdfjs/N3.js/issues/450; but have no responses.

Could you run tests on any codebases you're concerned about for yourself and then approve this PR if there are no issues?

Comment (tpluscode) [reply types/n3/index.d.ts:126]: Will see to it after the weekend

Comment (tpluscode) [reply types/n3/index.d.ts:126]: Hey. I did not see any issues but it also seems that I did not find any usages of the data factory directly fro n3.

**Review by tpluscode — APPROVED** [review]

**Comment by jeswr** [comment]
Ready to merge

File: types/n3/index.d.ts (hunk 1 above)
Comment (jeswr) [reply types/n3/index.d.ts:126]: Thanks for looking!

**Pull request merged**